lark/parsers/earley.py

```diff
@@ -299,7 +299,7 @@ def parse(self, stream, start):
         solutions = [n.node for n in columns[-1] if n.is_complete and n.node is not None and n.s == start_symbol and n.start == 0]
         if not solutions:
             expected_terminals = [t.expect for t in to_scan]
-            raise UnexpectedEOF(expected_terminals)
+            raise UnexpectedEOF(expected_terminals, state={i.s for i in to_scan})
```

Comment:
Wouldn't a `frozendict` be better?

Reply:
Also, I trust you that this makes sense. I do not know enough about earley to judge this.

Reply:
I don't think it matters, since no hashing is going to take place.

It's basically just a set of the next terminals to consider. It's somewhat similar to LALR state, but doesn't behave exactly the same. Neither of them are a sound and well-understood solution, they're both just a hack that should usually work.